CONTRIBUTING.md

```diff
@@ -1,10 +1,66 @@
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, on separate lines.
+
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
+
+Please run a local build test. Make sure there are no errors. After opening a pull request, keep an eye on the CI checks for quality or compile issues and fix them as required.
+
+If there are conflicts with jamulussoftware/jamulus showing when you go to raise the pull request, resolve those locally:
+~~~
+git checkout master ;# checkout your local master
+git pull upstream master --tags ;# refresh from jamulussoftware/jamulus (i.e. use your upstream remote name)
+git push --tags --force ;# update your fork's master
+git checkout - ;# back to your working branch
+git rebase master ;# replay your changes onto the latest code
+git push --force ;# update your working branch on your fork
+~~~
+
+### Supported platforms
+We support the following platforms and versions:
+- Windows 7 or higher
+- macOS <!-- Which versions? -->
+- Ubuntu Linux 18.04 or later, Debian 10 or later, most Linux flavors with recent enough Qt versions
+<!-- Do we support BSD? I think a recent discussion was about FreeBSD? -->
+<!-- Should we already list Android? If so, what platforms/versions? -->
+
+Please try to avoid breaking any of them by introducing platform-specific code. Check to see if any newly introduced Qt calls are available in the minimum supported Qt version, which is currently 5.9. Note that code _style_ in a file may be Qt 4.x, and while you should normally stick to existing style if making large-scale changes, then updating to Qt 5.9 style is encouraged.
+
+### Dependencies
+If your code requires new dependencies, be sure that those are available on all supported platforms and that the introduction of those dependencies has been discussed and accepted.
+
+### Ownership
+The submitter of an Issue or a PR is responsible for its care and feeding, answering all questions directed at them, and making agreed amends if needed. 
+
+So as to reduce effort for others in understanding a PR or an Issue, authors are strongly encouraged to update their initial posts/PR descriptions or title to reflect the current state of play, amends, enhancements, outstanding issues etc. Admins reserve the right to do this as they see fit.
+
+### Documentation/Acknowledgements
+Each new feature or bug fix must be documented in the ChangeLog. Please provide a single-sentence suggestion for that as part of your pull request description. Do not modify the ChangeLog file as part of your PR as it will lead to conflicts.
+
+If you are a first-time contributor/translator, please add yourself to the contributor/translator list in the About dialog of the Jamulus software (see in `src/util.cpp` in the constructor function `CAboutDlg::CAboutDlg()`).
```

Comment:
Just thinking out loud: Adding a link would break if we make changes to this file. Nevertheless, new contributors would need to scroll a lot here.

Reply:
Also thought about it. We could add a permalink to a specific revision to point to the right place. This will continue working, but might be confusing on the other hand. :(

Reply:
Now that I've done this https://github.com/jamulussoftware/jamuluswebsite/issues/314 I think I'll merge this too for consistency, and raise an issue about this point somewhere.